packages/lesswrong/server/repos/UsersRepo.ts

```diff
@@ -582,6 +582,51 @@ class UsersRepo extends AbstractRepo<"Users"> {
     `, [displayName]);
     return result.isDisplayNameTaken;
   }
+  
+  /**
+   * Returns a list of users who haven't read a post in over 3 months
+   * and who we want to email a feedback survey to.
+   *
+   * This excludes admins, deleted/deactivated users, flagged or purged or removed from queue users,
+   * and users who have already been sent this email.
+   */
+  async getInactiveUsersToEmail(limit: number): Promise<DbUser[]> {
+    return this.manyOrNone(`
+      -- UsersRepo.getInactiveUsersToEmail
+      SELECT
+        u.*
+      FROM public."Users" AS u
+      LEFT JOIN (
+        SELECT "userId", MAX("lastUpdated") AS max_last_updated
+        FROM "ReadStatuses"
+        WHERE "isRead" IS TRUE
+        GROUP BY "userId"
+      ) AS rs ON u._id = rs."userId"
+      WHERE
+        u."inactiveSurveyEmailSentAt" IS NULL
+        AND u."isAdmin" IS NOT TRUE
+        AND u.deleted IS NOT TRUE
+        AND u."deleteContent" IS NOT TRUE
+        AND u."sunshineFlagged" IS NOT TRUE
+        AND (
+          u."reviewedByUserId" IS NOT NULL
```

Comment:
You could also filter out snoozed users, but that may end up missing out on real users.